te-json-examples/te-json-examples.txt

```diff
@@ -0,0 +1,140 @@
+primary paths
+
+POST /restconf/data/ietf-te:te/tunnels HTTP/1.1
+    Host: example.com
+    Accept: application/yang-data+json
+ 
+{
+  "ietf-te:tunnel": [
+    {
+      "name": "Example_LSP_Tunnel",
+      "encoding": "te-types:lsp-encoding-packet",
+      "admin-state": "te-types:tunnel-state-up",
+      "source": "10.0.0.1",
+      "destination": "10.0.0.2",
+      "bidirectional": "false",
+      "signaling-type": "te-types:path-setup-rsvp"
+    }
+  ]
+}
+
+A.3 Create Global Constraint Policy: Path constraint 
+
+In the following example a the creation of a Named Path Constratint policy is requested. This
+NPC can be used later in furhter requests. In the example an NCP with 3 maximum hops is 
+created.
+
+
+POST /restconf/data/ietf-te:te/globals/named-path-constraints HTTP/1.1
+    Host: example.com
+    Accept: application/yang-data+json
+
+{
+  "ietf-te:named-path-constraint": {
+          "name": "max-hop-3",
+          "path-metric-bounds": {
+            "path-metric-bound": {
+              "metric-type": "te-types:path-metric-hop",
+              "upper-bound": "3"
+    }
+   }
+  }
+}
+
+A.4 Create tunnel with a constraint
+
+A.4.1 Create tunnel with a constraint using a NCP
+
+In the following example a RSVP-TE signaled Tunnel using a named path-constraint.
+We will reuse the NPC defined in the previous step.
+
+
+
+POST /restconf/data/ietf-te:te/tunnels HTTP/1.1
+    Host: example.com
+    Accept: application/yang-data+json
+
+{
+  "ietf-te:ietf-tunnel": [
+    {
+      "name": "Simple_LSP",
+      "encoding": "te-types:lsp-encoding-packet",
+      "description": "Simple_LSP_with_named_path",
+      "admin-state": "te-types:tunnel-state-up",
+      "source": "10.0.0.1",
+      "destination": "10.0.0.2",
+      "signaling-type": "path-setup-rsvp",
+      "bidirectional": "false",
+      "p2p-primary-paths": [
+        {
+          "p2p-primary-path": {
+            "name": "Simple_LSP_1",
+            "use-path-computation": "True",
+            "named-path-constraint": "path-metric-delay-minimum"
+          }
+        }
+      ]
+    }
+  ]
+}
+
+A.4.1 Create tunnel with indicating the constraint
+In the following example a RSVP-TE signaled Tunnel in which the 
+constraint is explicetly indicated. 
+{
+  "ietf-te:tunnel": [
+    {
+      "name": "Simple_LSP",
+      "encoding": "te-types:lsp-encoding-packet",
+      "provisioning-state": "te-types:tunnel-state-up",
+      "source": "10.0.0.1",
+      "destination": "10.0.0.2",
+      "bidirectional": "false",
+      "signaling-type": "te-types:path-setup-rsvp",
+      "p2p-primary-paths": {
+        "p2p-primary-path": [
+          {
+            "name": "path1",
+            "path-metric-bounds": {
+              "path-metric-bound": {
+                "metric-type": "te-types:path-metric-delay-average",
+                "upper-bound": "3"
+              }
+            }
+          }
+        ]
+      }
+    }
+  ]
+}
+
```

Comment:
Thanks Oscar. Can we please also have an operation get state for the tunnel- For example, to show the computed path properties and per LSP information.

Reply:
Added a query to get the path of a tunnel 
GET  /restconf/data/ietf-te:te/tunnels/tunnel="Example_LSP_Tunnel_A_4_1"/p2p-primary-paths/

Please check